internal/util/testutil/equal.go

```diff
@@ -166,20 +166,37 @@ func equalArrays(v1, v2 *types.Array) bool {
 	return true
 }
 
-// equalScalars compares BSON scalar values in a way that is useful for tests:
+// EqualScalars compares BSON scalar values in a way that is useful for tests:
 //  * float64 NaNs are equal to each other;
 //  * time.Time values are compared using Equal method.
-func equalScalars(v1, v2 any) bool {
+func EqualScalars(v1, v2 any) bool {
 	switch s1 := v1.(type) {
 	case float64:
 		s2, ok := v2.(float64)
 		if !ok {
-			return false
+			s3, ok := v2.(float32)
```

Comment:
No, that's a function is a version for assert.Equal that can compare NaNs. For testing. We don't want our testing helper to be smart about types.

Reply:
For the same reason we don't want `assert.Equal(t, 0, 0.0)` to pass. In tests, we want the comparison to be exact and without type conversions. In finding data we want to do automatic type conversions for numbers, for example, just like MongoDB would do.

Reply:
ok, thank you, that was not visible and clear (RFC for NaN comparison)

Reply:
Now let's revert all changes in that function – unexport it, etc.

Reply:
done